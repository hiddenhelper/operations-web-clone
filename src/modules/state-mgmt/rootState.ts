import { History } from 'history';

import { IState as IAuthState } from './auth/state';
import { IState as IGeneralState } from './general/state';
import { IState as IClientState } from './client/state';
import { IState as IProjectState } from './project/state';
import { IState as IUserState } from './user/state';
import { IState as IAccessControlSystemState } from './access-control-system/state';
import { IState as IBadgePrintingSystemState } from './badge-printer-system/state';
import { IState as IWorkerState } from './worker/state';
import { IState as ICertificationState } from './certification/state';
import { IState as IFileState } from './file/state';
import { IState as IBadgeState } from './badge/state';
import { IState as IStatisticsState } from './statistics/state';
import { IState as ITrainingState } from './training/state';
import { IState as IPaymentState } from './payment/state';
import { IState as IInvoiceState } from './invoice/state';
import { ApiService } from '../services/ApiService';
import { AuthService } from '../services/AuthService';
import { PrinterService } from '../services/PrinterService';
import { FileService } from '../services/FileService';
import { Logger } from '../services/Logger';
import { IProcoreState } from './procore/state';
import { IState as IProjectNewState } from './project-new/state';

export interface IAction {
  type: string;
  payload: any;
}

export interface IRootState {
  auth: IAuthState;
  user: IUserState;
  client: IClientState;
  project: IProjectState;
  accessControlSystem: IAccessControlSystemState;
  badgePrinterSystem: IBadgePrintingSystemState;
  worker: IWorkerState;
  certification: ICertificationState;
  training: ITrainingState;
  invoice: IInvoiceState;
  file: IFileState;
  badge: IBadgeState;
  statistics: IStatisticsState;
  payment: IPaymentState;
  general: IGeneralState;
  router?: { location: { pathname: string; search: string; hard: string; key: string }; action: string };
  procore: IProcoreState;
  projectNew: IProjectNewState;
}

export interface IEpicDependencies {
  history: History;
  apiService: ApiService;
  authService: AuthService;
  printerService: PrinterService;
  fileService: FileService;
  logger: typeof Logger;
}
