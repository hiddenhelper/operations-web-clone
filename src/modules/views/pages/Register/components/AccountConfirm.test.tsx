import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import AccountConfirm from './AccountConfirm';
import { createMockStore } from 'redux-test-utils';
import { getInitialState } from '../../../../../test/rootState';
import { Provider } from 'react-redux';

describe('Account Confirm', () => {
  it('should render loading', () => {
    const props = {
      email: 'test',
      loadingStatus: {
        isLoading: true,
        hasError: false,
        error: null,
      },
      onCreateAccount: jest.fn(),
    };
    const { container } = render(
      <Provider store={createMockStore(getInitialState()) as any}>
        <MemoryRouter>
          <AccountConfirm {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
