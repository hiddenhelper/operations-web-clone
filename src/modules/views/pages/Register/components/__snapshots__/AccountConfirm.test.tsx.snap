// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Confirm should render loading 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-52 makeStyles-box-43"
  >
    <div
      class="makeStyles-logoWrapper-44"
    >
      <a
        href="/"
      >
        <svg
          class="makeStyles-loginLogo-45"
          height="50px"
          version="1.1"
          viewBox="0 0 126 50"
          width="126px"
          xmlns="https://www.w3.org/2000/svg"
        >
          <title>
            FCA Logo
          </title>
          <g
            fill="none"
            fill-rule="evenodd"
            id="atoms/branding/logos/icon.fc.logo.2"
            stroke="none"
            stroke-width="1"
          >
            <path
              d="M90.2508624,36.0363855 L92.6338777,42.2340626 L92.6698039,42.2340626 L94.7860868,36.0363855 L97.0091144,36.0363855 L92.9719454,46.3894173 C92.3674039,47.9341914 91.6201916,48.8221914 89.7352317,48.8221914 C89.4684994,48.8221914 89.1915902,48.8021591 88.9123548,48.7546914 L88.6326348,48.6978043 L88.8101979,46.8510947 C89.094764,46.9396108 89.3795886,46.9754818 89.6106532,46.9754818 C90.4044532,46.9754818 90.6942411,46.6354455 90.992538,45.9557187 L91.1223947,45.6436109 L91.5493731,44.5605141 L87.8854225,36.0363855 L90.2508624,36.0363855 Z M23.3323559,0.25824515 C36.1983122,0.25824515 46.6652287,10.7090834 46.6652287,23.5547602 C46.6652287,36.4006951 36.1983122,46.8515334 23.3323559,46.8515334 C10.4669165,46.8515334 5.89750471e-13,36.4006951 5.89750471e-13,23.5547602 C5.89750471e-13,10.7090834 10.4669165,0.25824515 23.3323559,0.25824515 Z M122.048428,35.7881016 C123.198905,35.7881016 124.396404,36.1625724 125.112772,36.9263986 L125.249474,37.0843596 L123.862311,38.3093918 C123.489092,37.8123596 122.777548,37.386037 121.994926,37.386037 C121.319049,37.386037 120.625597,37.6348112 120.625597,38.3093918 C120.625597,38.928018 121.2085,39.1214539 122.27066,39.3791658 L122.475406,39.4283595 C123.737991,39.730295 125.231898,40.298295 125.231898,41.9677143 C125.231898,44.0453916 123.542594,44.8090044 121.674951,44.8090044 C120.44112,44.8090044 119.160494,44.3719329 118.330986,43.5709375 L118.171246,43.4059077 L119.576243,42.0918433 C120.038631,42.6957142 120.839086,43.1573916 121.763862,43.1573916 C122.386237,43.1573916 123.133191,42.9086174 123.133191,42.1450046 C123.133191,41.4348111 122.475406,41.2216498 121.176896,40.9199724 L120.874517,40.8429699 C119.720338,40.5286912 118.651468,39.9751053 118.651468,38.5050047 C118.651468,36.6048758 120.358865,35.7881016 122.048428,35.7881016 Z M114.131932,35.7879468 C115.298886,35.7879468 116.513317,36.2250183 117.217255,36.9673255 L117.35107,37.1198177 L115.892572,38.3982693 C115.555021,37.9187854 114.825901,37.5639467 114.131932,37.5639467 C112.566689,37.5639467 111.784326,38.9667854 111.784326,40.2983982 C111.784326,41.6302691 112.584781,43.0331078 114.203009,43.0331078 C114.80123,43.0331078 115.458406,42.8127932 115.880345,42.3993083 L115.999575,42.2694949 L117.333237,43.6191722 C116.62195,44.3827851 115.448277,44.8088496 114.185434,44.8088496 C111.660006,44.8088496 109.56104,43.1396884 109.56104,40.2983982 C109.56104,37.5280757 111.624597,35.7879468 114.131932,35.7879468 Z M79.3544346,35.7879468 C82.3259299,35.7879468 83.1159541,37.5642446 83.1586727,39.2285331 L83.1605391,39.3750434 L83.1605391,44.5603335 L81.1685761,44.5603335 L81.1685761,43.4769787 L81.115333,43.4769787 C80.6883545,44.2227851 79.6746684,44.7734948 78.465327,44.7734948 C77.0959978,44.7734948 75.263764,44.1164625 75.263764,42.1278175 C75.263764,39.7151853 78.1531746,39.2740789 80.7294548,39.2359998 L81.1331668,39.2331079 L81.1331668,39.0202047 C81.1331668,37.9722048 80.2438007,37.4573661 79.1768715,37.4573661 C78.3394562,37.4573661 77.5324322,37.8204033 76.9796187,38.3092002 L76.8468409,38.4338822 L75.744244,37.1376241 C76.704687,36.2496242 78.0561824,35.7879468 79.3544346,35.7879468 Z M101.897423,33.5682308 L101.897423,36.0363597 L104.03154,36.0363597 L104.03154,37.7413919 L101.897423,37.7413919 L101.897423,41.7013917 C101.897423,42.4650046 102.074986,43.0151981 103.017595,43.0151981 C103.245455,43.0151981 103.530052,42.9811749 103.762213,42.9040115 L103.924795,42.8379078 L103.995614,44.507327 C103.622395,44.6490045 103.053522,44.7377786 102.573042,44.7377786 C100.389636,44.7377786 99.8350277,43.443543 99.8004798,41.8816825 L99.7987155,41.7191981 L99.7987155,37.7413919 L98.3045495,37.7413919 L98.3045495,36.0363597 L99.7987155,36.0363597 L99.7987155,33.5682308 L101.897423,33.5682308 Z M107.763724,36.0366436 L107.763724,44.5605141 L105.629091,44.5605141 L105.629091,36.0366436 L107.763724,36.0366436 Z M70.8204485,35.7879984 C72.9693752,35.7879984 73.9053988,37.380774 73.9654127,39.0592773 L73.9685099,39.2331595 L73.9685099,44.5603851 L71.8341346,44.5603851 L71.8341346,39.8367724 L71.8287907,39.5794903 C71.787678,38.6235072 71.4887008,37.5639983 70.2334824,37.5639983 C68.9407477,37.5639983 68.2985563,38.6361827 68.2451574,39.723817 L68.2415194,39.8721272 L68.2415194,44.5603851 L66.1074026,44.5603851 L66.1074026,36.0365145 L68.1347748,36.0365145 L68.1347748,37.4039983 L68.1704425,37.4039983 C68.5439194,36.5516113 69.5576055,35.7879984 70.8204485,35.7879984 Z M86.5902977,31.1357405 L86.5902977,44.5605141 L84.4559224,44.5605141 L84.4559224,31.1357405 L86.5902977,31.1357405 Z M59.3875838,31.9877663 L64.8119159,44.5604109 L62.2508206,44.5604109 L61.0771468,41.6837659 L55.6171471,41.6837659 L54.4788825,44.5604109 L51.9712888,44.5604109 L57.4491224,31.9877663 L59.3875838,31.9877663 Z M42.0263612,15.8391477 L41.0158876,15.9562954 C30.2435832,17.2528368 14.505171,20.660146 7.72489812,27.7191471 C6.42070125,29.0814697 5.54399975,30.4623728 4.89448594,31.8525663 C8.07175349,38.8626305 15.1383501,43.7547593 23.3323559,43.7547593 C34.4878138,43.7547593 43.5636904,34.6930823 43.5636904,23.5547602 C43.5636904,20.8231475 43.0149765,18.2179863 42.0263612,15.8391477 Z M81.1331668,40.6715595 L80.3442239,40.6745676 C79.0294783,40.7012302 77.3981393,40.9244436 77.3981393,42.0919465 C77.3981393,42.9087207 78.162927,43.1574948 78.8920469,43.1574948 C80.2871324,43.1574948 81.0638351,42.3939088 81.1287373,41.1463343 L81.1331668,40.9732369 L81.1331668,40.6715595 Z M58.3560639,34.6517662 L56.3819347,39.7655724 L60.2945253,39.7655724 L58.3560639,34.6517662 Z M24.5498647,25.6531859 C25.6503595,25.6531859 26.8337089,26.0259091 27.5579651,26.7331689 L27.7059385,26.8898311 L26.3244616,28.2689278 C25.9432308,27.745573 25.2645109,27.4962826 24.6214586,27.4962826 C23.204314,27.4962826 22.1921787,28.58996 22.1921787,30.0764116 C22.1921787,31.610347 23.1805356,32.6567985 24.5855324,32.6567985 C25.3171509,32.6567985 25.9084714,32.3569007 26.2770743,31.9222874 L26.3720185,31.8005405 L27.7891631,33.1321534 C27.0507385,33.9884114 25.9313416,34.5233791 24.5141971,34.5233791 C21.9182095,34.5233791 19.976905,32.7161534 19.976905,30.0764116 C19.976905,27.3773149 21.9776556,25.6531859 24.5498647,25.6531859 Z M106.678444,31.8636115 C107.461066,31.8636115 108.065607,32.4143211 108.065607,33.1423211 C108.065607,33.870063 107.461066,34.4205146 106.678444,34.4205146 C105.896081,34.4205146 105.326949,33.8169017 105.326949,33.1423211 C105.326949,32.4674824 105.896081,31.8636115 106.678444,31.8636115 Z M34.4135062,25.8789666 L37.914626,34.2975469 L35.6517953,34.2975469 L34.9968538,32.6449018 L31.7340355,32.6449018 L31.1147617,34.2975469 L28.8994879,34.2975469 L32.4362755,25.8789666 L34.4135062,25.8789666 Z M18.6928681,25.8790182 L18.6928681,27.6150181 L15.1558221,27.6150181 L15.1558221,29.351018 L18.4186404,29.351018 L18.4186404,31.0276631 L15.1558221,31.0276631 L15.1558221,34.2975985 L13.1431822,34.2975985 L13.1431822,25.8790182 L18.6928681,25.8790182 Z M33.3892231,28.1261923 L32.3176416,31.0154825 L34.4372846,31.0154825 L33.3892231,28.1261923 Z M58.8007211,16.9491089 C60.3549795,16.9491089 62.1823871,17.4938695 63.2152826,18.6747704 L63.3716132,18.8665282 L61.6639579,20.2162055 C61.0059148,19.3106572 59.8144072,18.9199475 58.8007211,18.9199475 C56.2039581,18.9199475 54.4433182,20.9266571 54.4433182,23.5191731 C54.4433182,26.2538827 56.1150473,28.1540116 58.7118103,28.1540116 C59.9788372,28.1540116 60.9959842,27.6389159 61.6456944,26.8575116 L61.7707025,26.6977536 L63.5670101,27.9586568 C62.4644132,29.343689 60.8283517,30.1607212 58.6939765,30.1607212 C54.763552,30.1607212 51.9711337,27.4438181 51.9711337,23.5191731 C51.9711337,19.4882056 54.9235397,16.9491089 58.8007211,16.9491089 Z M69.5599575,21.0687991 C72.1388866,21.0687991 74.2019265,22.8623474 74.2019265,25.5438956 C74.2019265,28.2249278 72.1388866,30.0897019 69.5599575,30.0897019 C66.9812869,30.0897019 64.9179885,28.2249278 64.9179885,25.5438956 C64.9179885,22.8623474 66.9812869,21.0687991 69.5599575,21.0687991 Z M102.922171,21.0687733 C105.501101,21.0687733 107.56414,22.8623216 107.56414,25.5438698 C107.56414,28.224902 105.501101,30.0896761 102.922171,30.0896761 C100.343242,30.0896761 98.2802024,28.224902 98.2802024,25.5438698 C98.2802024,22.8623216 100.343242,21.0687733 102.922171,21.0687733 Z M88.357761,18.8489024 L88.357761,21.3172894 L90.4918778,21.3172894 L90.4918778,23.0220635 L88.357761,23.0220635 L88.357761,26.9820633 C88.357761,27.7454181 88.535324,28.2961278 89.4781917,28.2961278 C89.7056378,28.2961278 89.9903177,28.2619394 90.2225278,28.1845778 L90.3851332,28.1183213 L90.4562101,29.7877406 C90.0827332,29.9296761 89.5138594,30.0187083 89.0333794,30.0187083 C86.849974,30.0187083 86.2953655,28.7242315 86.2608176,27.162355 L86.2590534,23.0220635 L84.7648873,23.0220635 L84.7648873,21.3172894 L86.2590534,21.3172894 L86.2590534,18.8489024 L88.357761,18.8489024 Z M96.8135624,21.0687991 C96.9337471,21.0687991 97.0838809,21.0788152 97.2114074,21.0987386 L97.3294516,21.1219603 L97.3294516,23.0752506 C97.0448855,23.0042828 96.867064,22.9864764 96.6714086,22.9864764 C94.8191256,22.9864764 94.4056963,24.5385408 94.3785282,25.2065908 L94.3770456,29.8409277 L92.2429288,29.8409277 L92.2429288,21.3173152 L94.2881349,21.3173152 L94.2881349,22.7379603 L94.3238026,22.7379603 C94.7861902,21.7611861 95.6931317,21.0687991 96.8135624,21.0687991 Z M111.049546,16.4162573 L111.049546,29.8410309 L108.915171,29.8410309 L108.915171,16.4162573 L111.049546,16.4162573 Z M80.2658992,21.0687474 C82.4150757,21.0687474 83.3508661,22.661523 83.4108642,24.3400263 L83.4139606,24.5139086 L83.4139606,29.8408761 L81.2795853,29.8408761 L81.2795853,25.1175215 C81.2795853,24.1053925 81.0485207,22.8444893 79.6789331,22.8444893 C78.3861984,22.8444893 77.744007,23.9166737 77.690608,25.0045338 L77.6869701,25.1528763 L77.6869701,29.8408761 L75.5531117,29.8408761 L75.5531117,21.3172636 L77.5802255,21.3172636 L77.5802255,22.6844893 L77.6158931,22.6844893 C77.9893701,21.8323603 79.0030562,21.0687474 80.2658992,21.0687474 Z M102.922171,22.8089022 C101.232608,22.8089022 100.449987,24.2117409 100.449987,25.5438698 C100.449987,26.8754827 101.232608,28.3139342 102.922171,28.3139342 C104.611734,28.3139342 105.394097,26.8754827 105.394097,25.5438698 C105.394097,24.2117409 104.611734,22.8089022 102.922171,22.8089022 Z M69.5599575,22.808928 C67.8703945,22.808928 67.0880315,24.2117667 67.0880315,25.5438956 C67.0880315,26.8755085 67.8703945,28.31396 69.5599575,28.31396 C71.249779,28.31396 72.032142,26.8755085 72.032142,25.5438956 C72.032142,24.2117667 71.249779,22.808928 69.5599575,22.808928 Z M23.3323559,3.3550192 C12.1771564,3.3550192 3.10153832,12.4164382 3.10153832,23.5547602 C3.10153832,24.2205667 3.13776429,24.8776195 3.20267328,25.5265795 L3.256,26.0102451 L3.39601085,25.7130913 C6.05372305,20.2932045 14.284444,17.9582075 20.8405283,16.7994057 C19.6578084,16.1361799 18.8552853,14.87218 18.8552853,13.4218575 C18.8552853,11.2806963 20.5983498,9.54701893 22.7430636,9.54701893 C24.8805404,9.54701893 26.6238634,11.2806963 26.6238634,13.4218575 C26.6238634,14.4510187 26.2224727,15.3831477 25.5654635,16.0809541 C30.9256971,15.3865025 36.4632353,15.0985026 41.8206258,15.3671477 C38.6676537,8.29798673 31.5695247,3.3550192 23.3323559,3.3550192 Z M71.0269851,6.34931585 C73.4100004,6.34931585 75.1530649,7.89434803 75.1530649,10.9663479 C75.1530649,11.0965845 75.1530649,11.218907 75.1477808,11.3437911 L75.1352311,11.5343479 L68.6437114,11.5343479 C68.7326222,12.7420898 69.835219,13.6300897 71.0804867,13.6300897 C72.0762872,13.6300897 72.7625083,13.2587154 73.2398064,12.66053 L73.3389235,12.5289285 L74.7617542,13.6654446 C73.9258896,14.8019606 72.7165481,15.3702187 71.0978036,15.3702187 C68.4658899,15.3702187 66.5095946,13.683251 66.5095946,10.9129285 C66.5095946,8.1606706 68.448056,6.34931585 71.0269851,6.34931585 Z M90.1539394,1.69700638 L90.1539394,15.1215219 L88.1263087,15.1215219 L88.1263087,13.8430703 L88.090641,13.8430703 C87.5036749,14.8198445 86.3478349,15.370038 85.1563273,15.370038 C82.5595643,15.370038 80.9589121,13.3101671 80.9589121,10.8242318 C80.9589121,8.3026835 82.6663089,6.34939326 85.1030842,6.34939326 C86.5808811,6.34939326 87.4244588,7.0464 87.8883478,7.58733984 L87.966321,7.68100611 L88.0195641,7.68100611 L88.0195641,1.69700638 L90.1539394,1.69700638 Z M60.0989992,2.54905795 L60.0989992,4.50260625 L54.2122794,4.50260625 L54.2122794,7.94750932 L59.7611899,7.94750932 L59.7611899,9.82983182 L54.2122794,9.82983182 L54.2122794,15.1214445 L51.9711595,15.1214445 L51.9711595,2.54905795 L60.0989992,2.54905795 Z M64.371394,6.59793519 L64.371394,15.1215477 L62.2372772,15.1215477 L62.2372772,6.59793519 L64.371394,6.59793519 Z M79.1233442,1.69677412 L79.1233442,15.1215477 L76.9887104,15.1215477 L76.9887104,1.69677412 L79.1233442,1.69677412 Z M85.5830473,8.08952222 C83.9291519,8.08952222 83.1286966,9.49236087 83.1286966,10.8242318 C83.1286966,12.1561027 83.9291519,13.5945542 85.5830473,13.5945542 C87.1482903,13.5945542 88.090641,12.1917156 88.090641,10.8595866 C88.090641,9.52797377 87.1482903,8.08952222 85.5830473,8.08952222 Z M71.0088928,7.94750932 C69.8012032,7.94750932 68.8212393,8.74367881 68.6604974,9.85406197 L68.6437114,10.0073802 L73.0189481,10.0073802 C73.0189481,8.90647702 72.3965728,7.94750932 71.0088928,7.94750932 Z M63.2864501,2.42503215 C64.0690716,2.42503215 64.6736131,2.97548374 64.6736131,3.70348371 C64.6736131,4.43148367 64.0690716,4.98193526 63.2864501,4.98193526 C62.5040871,4.98193526 61.9349548,4.37832238 61.9349548,3.70348371 C61.9349548,3.02864503 62.5040871,2.42503215 63.2864501,2.42503215 Z"
              fill="#000B48"
              id="Shape"
            />
          </g>
        </svg>
      </a>
    </div>
    <div
      class="makeStyles-formContainer-46 undefined"
    >
      <div
        class="makeStyles-formContainerWrapper-47"
      >
        <h1
          class="MuiTypography-root makeStyles-formTitle-48 makeStyles-confirmTitle-14 MuiTypography-h1"
          data-testid="form-title"
        >
          Complete your Account Creation
        </h1>
        <p
          class="MuiTypography-root makeStyles-formSubtitle-51 MuiTypography-body1"
        >
          Please create your password in order to complete the account creation for 
          <span
            class="makeStyles-accentText-12"
          >
            test
          </span>
        </p>
        <form
          autocomplete="off"
          novalidate=""
        >
          <div
            class="MuiFormControl-root makeStyles-formControl-67 makeStyles-loginLabels-65 "
          >
            <h1
              class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-alignLeft"
            >
              Password
               
            </h1>
            <div
              data-testid="confirm-password-wrapper"
            >
              <div
                aria-describedby="confirm-password-password"
                class="MuiFormControl-root MuiTextField-root makeStyles-textField-66 MuiFormControl-fullWidth"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                >
                  <div
                    class="MuiInputAdornment-root     makeStyles-loginIcons-64 MuiInputAdornment-positionStart"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                      />
                    </svg>
                  </div>
                  <input
                    aria-invalid="false"
                    autocomplete="off"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                    data-testid="password"
                    name="password"
                    placeholder="Password"
                    required=""
                    type="password"
                    value=""
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                  >
                    <button
                      aria-label="toggle-visibility"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                      data-testid="toggle-visibility-button"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          class=" makeStyles-inactiveIcon-61 makeStyles-inactiveIcon-61"
                        >
                          visibility-off.svg
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-69 MuiOutlinedInput-notchedOutline"
                    style="padding-left: 8px;"
                  >
                    <legend
                      class="PrivateNotchedOutline-legend-70"
                      style="width: 0.01px;"
                    >
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <p
            class="MuiTypography-root makeStyles-inputSubtitle-56 MuiTypography-body1"
          >
            Password must have at least 8 characters, one lowercase letter, one uppercase letter, one number and one symbol.
          </p>
          <div
            class="MuiFormControl-root makeStyles-formControl-67 makeStyles-loginLabels-65 "
          >
            <h1
              class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-alignLeft"
            >
              Confirm Password
               
            </h1>
            <div
              data-testid="confirm-password-wrapper"
            >
              <div
                aria-describedby="confirm-password-password"
                class="MuiFormControl-root MuiTextField-root makeStyles-textField-66 MuiFormControl-fullWidth"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                >
                  <div
                    class="MuiInputAdornment-root     makeStyles-loginIcons-64 MuiInputAdornment-positionStart"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                      />
                    </svg>
                  </div>
                  <input
                    aria-invalid="false"
                    autocomplete="off"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                    data-testid="confirm-password"
                    name="confirmPassword"
                    placeholder="Confirm Password"
                    required=""
                    type="password"
                    value=""
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                  >
                    <button
                      aria-label="toggle-visibility"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                      data-testid="toggle-visibility-button"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          class=" makeStyles-inactiveIcon-61 makeStyles-inactiveIcon-61"
                        >
                          visibility-off.svg
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-69 MuiOutlinedInput-notchedOutline"
                    style="padding-left: 8px;"
                  >
                    <legend
                      class="PrivateNotchedOutline-legend-70"
                      style="width: 0.01px;"
                    >
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="makeStyles-termsWrapper-1"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-76 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
              data-testid="multiple-checkbox"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-79"
                  data-indeterminate="false"
                  data-testid="accept-terms"
                  name="acceptTerms"
                  type="checkbox"
                  value="acceptTerms"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-iconRoot-74 "
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <div>
              By signing up you accept the
               
              <span
                class=""
                data-testid="controlled-button"
                tabindex="-1"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-legalButton-2 makeStyles-accentText-12"
                  data-testid="open-terms-and-conditions"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Terms and conditions
                  </span>
                </button>
                 
              </span>
              and
               
              <span
                class=""
                data-testid="controlled-button"
                tabindex="-1"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-legalButton-2 makeStyles-accentText-12"
                  data-testid="open-privacy-policy"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Privacy Policy.
                  </span>
                </button>
              </span>
            </div>
          </div>
          <div
            class="makeStyles-submitWrapper-8"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-loading-80 makeStyles-submit-9 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge Mui-disabled MuiButton-fullWidth Mui-disabled"
              data-testid="sign-up-btn"
              disabled=""
              tabindex="-1"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                <span
                  class="makeStyles-spinnerWrapper-81"
                  data-testid="button-spinner"
                >
                  <svg
                    class="indicator_4oezz7"
                    height="23.2"
                    width="23.2"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      stroke="rgba(0, 0, 0, 0.3843137254901961)"
                      stroke-linecap="round"
                      stroke-width="2.3"
                      transform="translate(11.6, 11.6)"
                    >
                      <line
                        style="opacity: 1;"
                        transform="rotate(0)"
                        x1="0"
                        x2="0"
                        y1="6.6"
                        y2="9.6"
                      />
                      <line
                        style="opacity: 0.8860544217687075;"
                        transform="rotate(-30)"
                        x1="0"
                        x2="0"
                        y1="6.6"
                        y2="9.6"
                      />
                      <line
                        style="opacity: 0.7721088435374149;"
                        transform="rotate(-60)"
                        x1="0"
                        x2="0"
                        y1="6.6"
                        y2="9.6"
                      />
                      <line
                        style="opacity: 0.6581632653061225;"
                        transform="rotate(-90)"
                        x1="0"
                        x2="0"
                        y1="6.6"
                        y2="9.6"
                      />
                      <line
                        style="opacity: 0.5442176870748299;"
                        transform="rotate(-120)"
                        x1="0"
                        x2="0"
                        y1="6.6"
                        y2="9.6"
                      />
                      <line
                        style="opacity: 0.4302721088435374;"
                        transform="rotate(-150)"
                        x1="0"
                        x2="0"
                        y1="6.6"
                        y2="9.6"
                      />
                      <line
                        style="opacity: 0.3163265306122449;"
                        transform="rotate(-180)"
                        x1="0"
                        x2="0"
                        y1="6.6"
                        y2="9.6"
                      />
                      <line
                        style="opacity: 0.3163265306122449;"
                        transform="rotate(-210)"
                        x1="0"
                        x2="0"
                        y1="6.6"
                        y2="9.6"
                      />
                      <line
                        style="opacity: 0.3163265306122449;"
                        transform="rotate(-240)"
                        x1="0"
                        x2="0"
                        y1="6.6"
                        y2="9.6"
                      />
                      <line
                        style="opacity: 0.3163265306122449;"
                        transform="rotate(-270)"
                        x1="0"
                        x2="0"
                        y1="6.6"
                        y2="9.6"
                      />
                      <line
                        style="opacity: 0.3163265306122449;"
                        transform="rotate(-300)"
                        x1="0"
                        x2="0"
                        y1="6.6"
                        y2="9.6"
                      />
                      <line
                        style="opacity: 0.3163265306122449;"
                        transform="rotate(-330)"
                        x1="0"
                        x2="0"
                        y1="6.6"
                        y2="9.6"
                      />
                    </g>
                  </svg>
                </span>
                <span
                  class="makeStyles-textWrapper-82"
                >
                  Creating Account...
                </span>
              </span>
            </button>
          </div>
        </form>
      </div>
    </div>
    <div
      class="makeStyles-legalLabelWrapper-49"
    >
      <p
        class="MuiTypography-root makeStyles-legalLabel-50 MuiTypography-body1"
      >
        Field Control Analytics © 2020. All Rights Reserved.
      </p>
    </div>
  </div>
</div>
`;
